source/MaterialXGenGlsl/EsslShaderGenerator.h

```diff
@@ -20,9 +20,17 @@ using EsslShaderGeneratorPtr = shared_ptr<class EsslShaderGenerator>;
 class MX_GENGLSL_API EsslShaderGenerator : public GlslShaderGenerator
 {
   public:
-    EsslShaderGenerator();
-
-    static ShaderGeneratorPtr create() { return std::make_shared<EsslShaderGenerator>(); }
+    /// Constructor.
+    /// Optionally pass in an externally created TypeSystem here, 
+    /// if you want to keep type descriptions alive after the lifetime
+    /// of the shader generator.
+    EsslShaderGenerator(TypeSystemPtr typeSystem = TypeSystem::create());
```

Comment:
@niklasharrysson @ld-kerley Since we're encouraging developers to use the associated `create` function for each ShaderGenerator subclass, I would recommend omitting default arguments for our ShaderGenerator _constructors_, leaving only the default arguments on our ShaderGenerator `create` functions.  This would simplify the code a bit, and would help to clarify which entry point developers are expected to use.

Does that sound like a reasonable simplification to you both?

Reply:
Sounds good! I’ll make that change 👍